### PR TITLE
fix(test): replace osm-config configmap with MeshConfig

### PR DESCRIPTION
**Pull request by allenlsy**
Signed-off-by: Allen Leigh <allenlsy@gmail.com>

**Description**:

This is a continuation of the #3198 PR. 

Fix e2e test failure after rebased to main branch.

Primary changes:

* Remove unsupported installation/grade arguments from tests

**Affected area**:

- Install                [x]
- Tests                  [x]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

`No`

### CODE REVIEW

File: pkg/configurator/validating_webhook.go [hunk 1]
```diff
@@ -1,416 +0,0 @@
-package configurator
```
Comment (shashankram) [inline pkg/configurator/validating_webhook.go:1]: Could you carve removal of the configuration webhook into a separate PR. A lot of the code here will be reused for https://github.com/openservicemesh/osm/issues/3281. I'd prefer to retain this and tweak this separately.

File: cmd/cli/install.go [hunk 2]
```diff
@@ -205,6 +186,7 @@ func (i *installCmd) run(config *helm.Configuration) error {
 		return err
 	}
 
+	log.Trace().Msgf("run(): start resolveValues()")
```
Comment (shashankram) [inline cmd/cli/install.go:189]: remove.

File: pkg/configurator/validating_webhook.go (hunk 1 above)
Comment (snehachhabria) [reply pkg/configurator/validating_webhook.go:1]: +1, this will also reduce the size of the current PR, making it easier to review

File: cmd/cli/install.go [hunk 3]
```diff
@@ -55,33 +56,27 @@ well as for adding a Kubernetes Namespace to the list of Namespaces a control
...
-	defaultEnableDebugServer             = false
-	defaultEnableEgress                  = false
-	defaultEnablePermissiveTrafficPolicy = false
-	defaultDeployPrometheus              = false
-	defaultEnablePrometheusScraping      = true
-	defaultDeployGrafana                 = false
-	defaultEnableFluentbit               = false
-	defaultDeployJaeger                  = false
-	defaultEnforceSingleMesh             = false
+	defaultCertificateManager      = "tresor"
```
Comment (shashankram) [inline cmd/cli/install.go:59]: Can the install cli changes be carved out into a separate PR? Since they only update charts, it doesn't matter to leave them here for a bit longer.

This is going to cause a lot of conflicts with https://github.com/openservicemesh/osm/pull/3255

File: cmd/cli/install_test.go [hunk 4]
```diff
@@ -773,29 +768,60 @@ func TestEnforceSingleMeshWithExistingMesh(t *testing.T) {
...
+		controllerLogLevel:      testControllerLogLevel,
+		enforceSingleMesh:       true,
 	}
 
 	err = install.run(config)
 	assert.NotNil(err)
 	assert.Equal(err, errAlreadyExists)
 }
 
+func TestReadMeshConfigFile(t *testing.T) {
```
Comment (shashankram) [inline cmd/cli/install_test.go:791]: Why is the CLI dealing with meshconfig parsing? Seems like stale code?

File: cmd/cli/trafficpolicy_check.go [hunk 5]
```diff
@@ -55,7 +58,7 @@ func newTrafficPolicyCheck(out io.Writer) *cobra.Command {
 		Use:   "check-pods SOURCE_POD DESTINATION_POD",
 		Short: "check-pods traffic policy",
 		Long:  trafficPolicyCheckDescription,
-		Args:  cobra.ExactArgs(2),
+		Args:  cobra.MinimumNArgs(2),
```
Comment (shashankram) [inline cmd/cli/trafficpolicy_check.go:61]: This should be as before: `ExactArgs`

File: pkg/configurator/crd_client.go [hunk 6]
```diff
@@ -1,6 +1,7 @@
 package configurator
```
Comment (shashankram) [inline pkg/configurator/crd_client.go:1]: Rename file to `client.go`.

**Review by shashankram — CHANGES_REQUESTED** [review]
Hey @allenlsy, I see a few issues, otherwise looks good.
1. It would be carve out removal of dead-code in a separate PR (such as code related to the validating webhook)
2. This change updates the functionality in a way that leaves the existing documentation regarding install time flags stale. Would you mind updating all the affected documentation (ex. egress, permissive mode etc.).

File: pkg/configurator/crd_client.go (hunk 6 above)
Comment (allenlsy) [reply pkg/configurator/crd_client.go:1]: Prefer to do it in a separate PR. So that in the code review here, we can look at a complete client implementation.

Comment (allenlsy) [reply pkg/configurator/crd_client.go:1]: Changed mind. Make sense to use `client.go` directly

File: cmd/cli/install.go (hunk 3 above)
Comment (allenlsy) [reply cmd/cli/install.go:59]: Yes.

we can keep the CI arguments in install.go. But they won't be added in osm-controller deployment

File: cmd/cli/install_test.go (hunk 4 above)
Comment (allenlsy) [reply cmd/cli/install_test.go:791]: This was an initial change, where osm CLI can take a MeshConfig manifest file.

Will remove.

File: pkg/configurator/validating_webhook.go (hunk 1 above)
Comment (allenlsy) [reply pkg/configurator/validating_webhook.go]: validating_webhook is recovered

File: cmd/cli/install.go (hunk 3 above)
Comment (allenlsy) [reply cmd/cli/install.go]: install.go reverted

File: cmd/cli/install.go [hunk 7]
```diff
@@ -159,6 +159,7 @@ func newInstallCmd(config *helm.Configuration, out io.Writer) *cobra.Command {
 				return errors.Errorf("Could not access Kubernetes cluster, check kubeconfig: %s", err)
 			}
 			inst.clientSet = clientset
+
```
Comment (shashankram) [inline cmd/cli/install.go:162]: nit: file can be ignored.

File: cmd/cli/mesh_upgrade.go [hunk 8]
```diff
@@ -23,18 +23,15 @@ const (
 const upgradeDesc = `
 This command upgrades an OSM control plane configuration by upgrading the
```
Comment (shashankram) [inline cmd/cli/mesh_upgrade.go:24]: Could you drop this change from this PR? Don't see it being necessary here.

**Review by shashankram — COMMENTED** [review]
Change looks good. Needs to be rebased.

**Comment by allenlsy** [comment]
@shashankram There is a separate task for the documentation https://github.com/openservicemesh/osm/issues/3040

File: cmd/cli/mesh_upgrade.go (hunk 8 above)
Comment (allenlsy) [reply cmd/cli/mesh_upgrade.go:30]: reverted

File: .github/workflows/main.yml [hunk 9]
```diff
@@ -193,7 +193,7 @@ jobs:
         if: ${{ github.event_name == 'pull_request' }}
         env:
           K8S_NAMESPACE: "osm-system"
-        run: go test ./tests/e2e -test.v -ginkgo.v -ginkgo.progress -installType=KindCluster -test.timeout 0 -test.failfast -ginkgo.failFast -ginkgo.focus='\[Tier 1\]\[Bucket ${{ matrix.bucket }}\]' -ginkgo.skip='Upgrade'
+        run: go test ./tests/e2e -test.v -ginkgo.v -ginkgo.progress -installType=KindCluster -test.timeout 0 -test.failfast -ginkgo.failFast -ginkgo.skip='Upgrade'
```
Comment (shashankram) [inline .github/workflows/main.yml:196]: Note to others: this is a temporary hack that needs to be reverted

**Review by shashankram — APPROVED** [review]

File: charts/osm/crds/meshconfig.yaml [hunk 10]
```diff
@@ -83,12 +83,16 @@ spec:
                       default: false
                     outboundIPRangeExclusionList:
                       description: Global list of IP address ranges to exclude from outbound traffic interception by the sidecar proxy.
-                      type: string
-                      pattern: ((?:\d{1,3}.){3}\d{1,3})\/(\d{1,2})(,((?:\d{1,3}.){3}\d{1,3})\/(\d{1,2}))*$
+                      type: array
+                      items:
+                        type: string
+                        pattern: ((?:\d{1,3}.){3}\d{1,3})\/(\d{1,2})$
```
Comment (snehachhabria) [inline charts/osm/crds/meshconfig.yaml:89]: why did the pattern change ?

Comment (allenlsy) [reply charts/osm/crds/meshconfig.yaml:89]: https://github.com/openservicemesh/osm/blob/main/pkg/apis/config/v1alpha1/mesh_config.go#L37

The CRD definition actually should be changed initially. After changing configmap to MeshConfig `outboundIPRangeExclusionList` does not need to be a comma separated string. So changing it to an array of CIDRs

**Comment by allenlsy** [comment]
Screenshot of all e2e tests passed.

Screen Shot 2021-05-04 at 4 11 51 PM

File: .github/workflows/main.yml (hunk 9 above)
Comment (allenlsy) [reply .github/workflows/main.yml]: Reverted

File: charts/osm/templates/validatingwebhook.yaml [hunk 11]
```diff
@@ -33,4 +33,4 @@ webhooks:
       resources:
         - configmaps
   sideEffects: None
-  admissionReviewVersions: ["v1"]
+  admissionReviewVersions: ["v1"]
```
Comment (snehachhabria) [inline charts/osm/templates/validatingwebhook.yaml:36]: nit: lets add back the new line

File: charts/osm/values.schema.json [hunk 12]
```diff
@@ -490,15 +377,7 @@
...
-                "enablePrivilegedInitContainer": {
-                    "$id": "#/properties/OpenServiceMesh/properties/enablePrivilegedInitContainer",
-                    "type": "boolean",
-                    "title": "The enablePrivilegedInitContainer schema",
-                    "description": "Indicates whether the init container for pods in the mesh should be privileged",
-                    "examples": [
-                        false
-                    ]
-                },
+
```
Comment (snehachhabria) [inline charts/osm/values.schema.json:380]: nit: remove new lint

File: charts/osm/values.yaml [hunk 13]
```diff
@@ -158,15 +117,11 @@ OpenServiceMesh:
...
   # -- Feature flags for experimental features
   featureFlags:
     # Enable extra Envoy statistics generated by a custom WASM extension
     enableWASMStats: false
 
     # Enable OSM's Egress policy API
     # If specified, fine grained control over Egress (external) traffic is enforced
-    enableEgressPolicy: false
\ No newline at end of file
+    enableEgressPolicy: false
```
Comment (snehachhabria) [inline charts/osm/values.yaml:127]: nit : lets add the new line back here

File: cmd/cli/install.go [hunk 14]
```diff
@@ -95,6 +95,7 @@ func newInstallCmd(config *helm.Configuration, out io.Writer) *cobra.Command {
...
```
Comment (snehachhabria) [inline cmd/cli/install.go:98]: nit : remove new line

File: cmd/cli/mesh_upgrade.go [hunk 15]
```diff
@@ -29,18 +29,15 @@ const (
...
 underlying Helm release.
-
```
Comment (snehachhabria) [inline cmd/cli/mesh_upgrade.go:32]: these changes are not needed, please can you revert

File: cmd/osm-controller/validate_test.go [hunk 16]
```diff
@@ -1,147 +0,0 @@
-package main
```
Comment (snehachhabria) [inline cmd/osm-controller/validate_test.go:1]: if this test file is being deleted shouldn't the `validate.go` file also be deleted in this PR ?

Comment (allenlsy) [reply cmd/osm-controller/validate_test.go:1]: Let me add back validate_test.go. osm-injector is still using methods in validate.go

**Review by snehachhabria — APPROVED** [review]
